internal/guidedremediation/matcher/matcher.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package matcher provides the interface for the vulnerability matcher used by guided remediation.
+package matcher
+
+import (
+	"context"
+
+	"github.com/google/osv-scalibr/extractor"
+)
+
+// Temporarily internal while migration is in progress.
+// Will need to be moved to publicly accessible location once external interface is created.
+
+// VulnerabilityMatcher interface provides functionality get a list of affecting vulnerabilities for each package in an inventory.
+type VulnerabilityMatcher interface {
+	MatchVulnerabilities(ctx context.Context, invs []*extractor.Inventory) ([][]*OSVRecord, error)
+}
+
+// OSVRecord is a representation of an OSV record.
+// TODO: replace with https://github.com/ossf/osv-schema/pull/333
+type OSVRecord struct {
```

Comment:
I should be able to do this soon

Reply:
internally we have a policy that all TODOs need to have a bug attached - can you create a github issue for moving Guided Remediation and refer to it here?

Reply:
I've now replaced this with `osvschema.Vulnerability`, so the TODO is done anyway.

Currently am in the process of getting `osvschema` into google3